apache/superset
# fix(explore): Dashboard filters on adhoc cols not working in Explore

### SUMMARY
When user applied a cross filter on an adhoc column and then tried to explore a chart targeted by that cross filter, the adhoc filter control would be display as `[Object object]`. This PR implements a proper translation of a filter based on adhoc column to an SQL adhoc filter

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:

<img width="360" alt="image" src="https://user-images.githubusercontent.com/15073128/221939470-ff5593b4-0e80-4569-8a43-83512e835d38.png">

<img width="439" alt="image" src="https://user-images.githubusercontent.com/15073128/221939505-6a127826-640c-4564-afa3-0d463fedd93b.png">

After:

<img width="359" alt="image" src="https://user-images.githubusercontent.com/15073128/221939673-c0f3eb85-4559-49b3-96fd-75af04595ad0.png">

<img width="445" alt="image" src="https://user-images.githubusercontent.com/15073128/221939747-66303873-3be3-4e31-b3af-a059c82cfc64.png">


### TESTING INSTRUCTIONS
1. Add a chart with grouping based on an adhoc column to a dashboard
2. Add another chart that uses the same dataset to the dashboard
3. Apply cross-filtering by clicking on a series related to the adhoc column in the first chart
4. Open the second chart in Explore by clicking on the chart's title in dashboard
5. Verify that the adhoc filter based on the cross-filter from the dashboard is displayed properly

### ADDITIONAL INFORMATION
